src/goto-programs/goto_contractor.cpp

```diff
@@ -192,8 +212,13 @@ goto_contractort::create_constraint_from_expr2t(irep_container<expr2t> expr)
 {
   ibex::NumConstraint *c;
   if(
-    is_arith_expr(expr) || is_constant_number(expr) || is_symbol2t(expr) ||
-    is_notequal2t(expr) || is_equality2t(expr))
+    is_arith_expr(get_base_object(expr)) ||
```

Comment:
This code looks very weird. Can you move it to a method and explain it?

Reply:
maybe moving the `get_base_object(expr)` calls to before this branch should make it more readable as well.

Reply:
Basically a list of unsupported operators. I moved it to a function and performed get_base_object before the checks.